### DISCUSSION TITLE
Vertx OTEL Tracing issues

### DISCUSSION THREAD

**Discussion by jimbogithub** [quarkusio/quarkus#41346] [Q&A] [answered]
OTEL Tracing in a Vertx Verticle appears to have two issues:
1. Trace IDs are not injected into logs.
2. Spans are not nested correctly.

Is there extra config required to get these working with a Verticle or are these known issues?

Reproducer available here: https://github.com/jimbogithub/vertx-tracing

- It's a simple echo server so start it up then connect with `telnet` or `nc` to exercise it.
- Note no trace ids in logs.
- Spans do appear in the dev-services Grafana (Tempo) but each is a separate Trace whereas I believe they (3 per echo) should be  under the same Trace.

[a comment by an automated account is not shown]

**Comment by brunobat** [comment]
Hi @jimbogithub. 
The Vert.x context propagation is required for spans to be identified as parents and also for the MDC logging context setup.
@cescoffier I wonder if there is an example somewhere on how to propagate context in an example like this.

**Comment by cescoffier** [comment, marked as answer]
We do not handle TCP servers automatically, because they are not managed. 
In this case, you must a duplicated context.

Here is an example:

https://github.com/quarkusio/quarkus/blob/main/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/options/HttpServerCommonHandlers.java#L68-90

**Reply by jimbogithub** [reply]
Thanks.  Applied to reproducer and all works as expected.  Similar approach works for my real world app.
